### PR TITLE
adding lzout.all

### PR DESCRIPTION
**Pull request by djarecka**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Summary

allowing for passing entire result object by using `lzout.all`

## Checklist

- [ ] All tests passing
- [x] I have added tests to cover my changes
- [ ] I have updated documentation (if necessary)
- [ ] My code follows the code style of this project 
(we are using `black`: you can `pip install pre-commit`, 
run `pre-commit install` in the `pydra` directory 
and `black` will be run automatically with each commit)

## Acknowledgment
- [x] I acknowledge that this contribution will be available under the Apache 2 license.

### CODE REVIEW

**Comment by djarecka** [comment]
@satra - at the end I decided to use `lzout.all` syntax, what do you think?

File: pydra/engine/specs.py [hunk 1]
```diff
@@ -152,7 +152,7 @@ def __init__(self, node, attr_type):
         self.field = None
 
     def __getattr__(self, name):
-        if name in self.fields:
+        if name in self.fields or name == "all":
```
Comment (satra) [inline pydra/engine/specs.py]: in general this is fine, but this makes `all` a reserved output name (no interface can have it as an output). could we use `all_` ?

File: pydra/engine/specs.py [hunk 2]
```diff
@@ -184,13 +184,24 @@ def get_value(self, wf, state_index=None):
                 if isinstance(result[0], list):
                     results_new = []
                     for res_l in result:
-                        res_l_new = [getattr(res.output, self.field) for res in res_l]
+                        if self.field == "all":
+                            res_l_new = [res.output.__dict__ for res in res_l]
```
Comment (satra) [inline pydra/engine/specs.py]:
```suggestion
                            res_l_new = [dc.asdict(res.output) for res in res_l]
```

File: pydra/engine/specs.py [hunk 3]
```diff
@@ -184,13 +184,24 @@ def get_value(self, wf, state_index=None):
...
+                        else:
+                            res_l_new = [
+                                getattr(res.output, self.field) for res in res_l
+                            ]
                         results_new.append(res_l_new)
                     return results_new
                 else:
-                    return [getattr(res.output, self.field) for res in result]
+                    if self.field == "all":
+                        return [res.output.__dict__ for res in result]
```
Comment (satra) [inline pydra/engine/specs.py]:
```suggestion
                        return [dc.asdict(res.output) for res in result]
```

File: pydra/engine/specs.py [hunk 4]
```diff
@@ -184,13 +184,24 @@ def get_value(self, wf, state_index=None):
...
+                    else:
+                        return [getattr(res.output, self.field) for res in result]
             else:
-                return getattr(result.output, self.field)
+                if self.field == "all":
+                    return result.output.__dict__
```
Comment (satra) [inline pydra/engine/specs.py]:
```suggestion
                    return dc.asdict(result.output)
```

**Comment by satra** [comment]
we should add this to the tutorials.

**Comment by djarecka** [comment]
yes, planning to add multiple things to tutorial, especially regarding `ShellTask` and `ContainerTask`

**Pull request merged**